File: hideapi/src/main/java/android/app/ActivityThread.java
package android.app;

import utils.Utils;

public class ActivityThread {
    public static ActivityThread currentActivityThread() {
        return Utils.throwStub();
    }

    public static Application currentApplication() {
        return Utils.throwStub();
    }

    public ContextImpl getSystemContext() {
        return Utils.throwStub();
    }
}

File: module/src/main/java/com/github/kr328/intent/compat/ActivityThread.kt
package com.github.kr328.intent.compat

import android.app.ActivityThread
import android.app.Application
import android.content.Context

val currentApplication: Application?
    get() = ActivityThread.currentApplication()

val currentActivityThread: ActivityThread?
    get() = ActivityThread.currentActivityThread()

val currentSystemContext: Context?
    get() = currentActivityThread?.systemContext

val requiredSystemContext: Context
    get() = currentSystemContext!!
